### ISSUE TITLE
Exception if perfdata contains a unit but the value is an empty string

### ISSUE THREAD

**Issue by ghost** [opsdis/monitor-exporter#25]
[2022-04-26 13:58:06,783] ERROR in app: Exception on request GET /metrics
Traceback (most recent call last):
  File "/home/mike/.local/lib/python3.9/site-packages/quart/app.py", line 1489, in handle_request
    return await self.full_dispatch_request(request_context)
  File "/home/mike/.local/lib/python3.9/site-packages/quart/app.py", line 1514, in full_dispatch_request
    result = await self.handle_user_exception(error)
  File "/home/mike/.local/lib/python3.9/site-packages/quart/app.py", line 964, in handle_user_exception
    raise error
  File "/home/mike/.local/lib/python3.9/site-packages/quart/app.py", line 1512, in full_dispatch_request
    result = await self.dispatch_request(request_context)
  File "/home/mike/.local/lib/python3.9/site-packages/quart/app.py", line 1557, in dispatch_request
    return await self.ensure_async(handler)(**request_.view_args)
  File "/home/mike/monitor-exporter-1.2.0/monitor_exporter/proxy.py", line 49, in get_metrics
    await asyncio.get_event_loop().create_task(monitor_data.get_perfdata())
  File "/home/mike/monitor-exporter-1.2.0/monitor_exporter/perfdata.py", line 147, in get_perfdata
    normilized_value, prometheus_key_with_labels = self.create_metric(check_command, labels,
  File "/home/mike/monitor-exporter-1.2.0/monitor_exporter/perfdata.py", line 174, in create_metric
    normilized_value, unit = Perfdata.normalize_to_unit(perf_value, perf_unit)
  File "/home/mike/monitor-exporter-1.2.0/monitor_exporter/perfdata.py", line 279, in normalize_to_unit
    return value / 1000.0, 'seconds'
TypeError: unsupported operand type(s) for /: 'str' and 'float'

I added 2 rows at the top of normalize_to_unit in monitor_exporter/perfdata.py. Might be better to return the value None or 0 than ''?

    def normalize_to_unit(value, unit):
        """Normalize the value to the unit returned.
        We use base-1000 for second-based units, and base-1024 for
        byte-based units. Sadly, the Nagios-Plugins specification doesn't
        disambiguate base-1000 (KB) and base-1024 (KiB).
        """
->      if value == '':
->          return value, ''
        if unit == '%':
            return value / 100, 'ratio'

**Comment by thenodon** [comment]
Thanks for the bug report. 
I think the best is to add a check if the value is a string, than nothing of the normalization should be done.
```
        if isinstance(value, str):
            return value, ''
```
This will work for a empty and none empty string value and be in line with the return statement if no normalization are done. The question is how this will end up in Prometheus. A metrics without any value does not make sense, and should probably be considered a bug from the plugin  where perfdata has no value. 
The options we have are:
- metrics should be dropped if value is a string, empty or not.
- Set the value to NaN which I **think** Prometheus can handle, but can lead to other complication in promql. 
I think the best is to enable a config option with dropping as default, plus a log message, and with the option to set it to NaN.
With the "fix" you implemented just returning it how does i looks like in Prometheus when you query on it?

**Comment by ghost** [comment]
I can't find a timeserie in Prometheus for the check causing this error but it has been returning faulty performance data all the time. We alse get error when trying to display the graph in Op5. Dropping the metric should be fine.

**Comment by msafdal** [comment]
I'm voting for the "drop metric" option.

Experiencing similar issues where the performance data in OP5 Monitor is broken - which leads to an empty value in the export which causes Prometheus to throw 'expected value after metric, got "MNAME"' while scraping the target.

Test against a host with broken performance data:

```
curl --silent http://localhost:9631/metrics?target=xxx.yyy.zzz | promtool check metrics
error while linting: text format parsing error in line 16: expected float as value, got ""
```

How the output looks like:

```
monitor_service_state{hostname="xxx.yyy.zzz", service="Obfuscated service", downtime="false", address="xx.xx.xx.xx", flapping="0", acknowledged="0"} 0
monitor_check_json_custom2_ratio{hostname="xxx.yyy.zzz", service="Obfuscated faulty service", downtime="false", address="xx.xx.xx.xx", flapping="0", acknowledged="0"} 
monitor_service_state{hostname="xxx.yyy.zzz", service="Obfuscated service", downtime="false", address="xx.xx.xx.xx", flapping="0", acknowledged="0"} 0
```

**Comment by thenodon** [comment]
Fix in release v1.3.0 and avaliable on pypi.org
